dbt/include/spark/macros/materializations/table.sql

```diff
@@ -17,15 +16,24 @@
     {{ adapter.drop_relation(old_relation) }}
   {%- endif %}
 
-  -- build model
-  {% call statement('main') -%}
-    {{ create_table_as(False, target_relation, sql) }}
-  {%- endcall %}
-  
+  {%- call statement('main', language=language) -%}
+    {{ create_table_as(False, target_relation, compiled_code, language) }}
+  {%- endcall -%}
+
   {% do persist_docs(target_relation, model) %}
 
   {{ run_hooks(post_hooks) }}
 
   {{ return({'relations': [target_relation]})}}
 
 {% endmaterialization %}
+
+
+{% macro py_complete_script(compiled_code, target_relation) %}
+{{ compiled_code }}
+
+# --- Autogenerated dbt code below this line. Do not modify. --- #
+dbt = dbtObj(spark.table)
```

Comment:
Where does `spark.table` come from? Should we use `databricks.table` in `dbt-databricks`?

Reply:
This is just a python function later used as `dbt_load_df_function` to load data [here](https://github.com/dbt-labs/dbt-core/blob/cf9d0200d7ad2ebbaa8fbd9cfca5c71eed6d4b66/core/dbt/include/global_project/macros/python_model/python.sql#L9-L12).

```python
def ref(*args,dbt_load_df_function):
    refs = {{ ref_dict | tojson }}
    key = ".".join(args)
    return dbt_load_df_function(refs[key])
```

This works with the current notebook submit solution. I haven't tested `databricks.table`. Do you think it will perform better than `spark.table`?

Reply:
Ah, got it, `spark` is the one on the notebook. Then it should be fine. Thanks.

Reply:
I think we can make the `spark` more explicit and thus not expect the notebook to magically insert this global variable by adding above `{{ compiled_code }}`:

```
from pyspark.sql import SparkSession
session = SparkSession.builder.enableHiveSupport().getOrCreate()
````